src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledDictionaryBuilder`2.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Microsoft.AspNetCore.Razor.PooledObjects;
+
+/// <summary>
+///  Wraps a pooled <see cref="ImmutableDictionary{TKey, TValue}.Builder"/> but doesn't allocate it until
+///  it's needed. Note: Dispose this to ensure that the pooled array builder is returned
+///  to the pool.
+/// </summary>
+internal ref struct PooledDictionaryBuilder<TKey, TValue>(ObjectPool<ImmutableDictionary<TKey, TValue>.Builder>? pool)
```

Comment:
Nit: Is there any harm in making this `IDisposable`, in case anyone looks at the source and misses the comment?

Reply:
It's a ref struct, so it can't implement IDisposable. For ref structs, the compiler supports the "disposable pattern" by looking for a public Dispose method.